File: php/features.php
<?php
declare(strict_types=1);

// TRexVpn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TRexVpnFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TRexVpnBaseFeature();
            case "test":
                return new TRexVpnTestFeature();
            default:
                return new TRexVpnBaseFeature();
        }
    }
}

File: py/trexvpn_sdk/features.py
# TRexVpn SDK feature factory

from trexvpn_sdk.feature.base_feature import TRexVpnBaseFeature
from trexvpn_sdk.feature.test_feature import TRexVpnTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TRexVpnBaseFeature(),
        "test": lambda: TRexVpnTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
